src/utils/SetupUtil.js

```diff
@@ -85,15 +86,31 @@ export default {
   },
 
   async nonNativeSetup () {
+    let hypervBoxVersion = null;
     let virtualBoxVersion = null;
     let machineVersion = null;
+    let startedHyperv = false;
+   
     while (true) {
       try {
         setupServerActions.started({started: false});
 
         // Make sure virtualBox and docker-machine are installed
         let virtualBoxInstalled = virtualBox.installed();
+        let hypervInstalled = hypervBox.installed();
         let machineInstalled = machine.installed();
+        
+        if (!hypervInstalled || !machineInstalled) {
+          router.get().transitionTo('setup');
+          if (!hypervInstalled) {
+            setupServerActions.error({error: 'Hyper-V is not installed. Please install it via the Docker Toolbox. Setup will continue and check for VirtualBox'});
+          } else {
+            setupServerActions.error({error: 'Docker Machine is not installed. Please install it via the Docker Toolbox.'});
+          }
+          
+          //do nothing
+        }
+        
         if (!virtualBoxInstalled || !machineInstalled) {
```

Comment:
On my machine VBox was never installed. ... So I get an error here. 

IMO we should have a single check for `machineInstalled` and then a second one ot check `if vbox else if hyperv`